nomt/src/bitbox/mod.rs

```diff
@@ -139,6 +139,14 @@ impl DB {
             ht_pages.push((pn, buf));
         }
 
+        if cfg!(debug_assertions) {
+            // Make sure that there are no duplicate pages.
+            let orig_len = ht_pages.len();
+            ht_pages.sort_unstable_by_key(|(pn, _)| *pn);
+            ht_pages.dedup_by_key(|(pn, _)| *pn);
+            assert_eq!(orig_len, ht_pages.len());
```

Comment:
I know this is not a critical piece of code but it can be quickly improved to perform the check in a single run:
```rust
if cfg!(debug_assertions) {
    let mut seen_pages = HashSet::new();
    for (pn, _) in &ht_pages {
        assert!(seen_pages.insert(*pn), "duplicate page");
    }
}
```